docs/source/developers/guide/step_by_step/building.rst

```diff
@@ -0,0 +1,37 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. _build-arrow:
+
+*************************
+Building Arrow package 🏋🏿‍♀️
+*************************
```

Comment:
How are we going to make this coexist with the already existing "build" guides? (IE: https://arrow.apache.org/docs/developers/python.html#building-on-linux-and-macos )

I'm not against repetition, but maybe if the current guides are not good enough it means we need to replace them instead of adding one more?

Reply:
PS: That does not mean we should not reference how to build arrow from the contributors guide. We don't want people having to read 4 different guides to learn how to do things, so there is still a value in having a section about how to build things in the contributors guide. We should just make sure we don't end up with two docs to maintain instead of one.

Reply:
What I want to do here is to connect build docs and add some beginner explanation. Will keep it short an link the existing docs
https://github.com/apache/arrow/pull/11677#discussion_r752418532